staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -608,6 +608,20 @@ func (c completedConfig) New(name string, delegationTarget DelegationTarget) (*G
 	if c.SharedInformerFactory != nil && !s.isPostStartHookRegistered(genericApiServerHookName) {
 		err := s.AddPostStartHook(genericApiServerHookName, func(context PostStartHookContext) error {
```

Comment:
Consider adding a second post start hook rather than modify this one. Adding stuff to this one could surprise people who have tuned healthz/livez/readyz. It'll also let people decide e.g. that waiting is optional, but starting is not.